fenix/app/src/main/java/org/mozilla/fenix/compose/LinkText.kt

```diff
@@ -95,15 +97,21 @@ fun LinkText(
         modifier = Modifier.clickable(
             enabled = context.isScreenReaderEnabled,
             role = Role.Button,
-            onClickLabel = linkTextState.text,
-            onClick = { linkTextState.onClick(linkTextState.url) },
+            onClickLabel = linkTextStates.first().text,
+            onClick = {
+                linkTextStates.first().let {
+                    it.onClick(it.url)
+                }
+            },
         ),
         onClick = {
             if (!context.isScreenReaderEnabled) {
                 val range: AnnotatedString.Range<String>? =
                     annotatedString.getStringAnnotations(URL_TAG, it, it).firstOrNull()
                 range?.let { stringAnnotation ->
-                    linkTextState.onClick(stringAnnotation.item)
+                    linkTextStates.first { state ->
```

Comment:
Also `firstOrNull`

Let's also extract this block inside the if statement as a function and add tests for this function. It seems they are simple kotlin data structures and no compose/android depedendencies so we could write pure kotlin tests.